virtual_ecosystem/models/animal/animal_communities.py

```diff
@@ -353,3 +363,58 @@ def inflict_non_predation_mortality_community(self, dt: timedelta64) -> None:
             if cohort.individuals <= 0:
                 cohort.is_alive = False
                 self.remove_dead_cohort(cohort)
+
+    def metamorphose(self, larval_cohort: AnimalCohort) -> None:
+        """This transforms a larval status cohort into an adult status cohort.
+
+        This method takes an indirect developing cohort in its larval form,
+        inflicts a mortality rate, and creates an adult cohort of the correct type.
+
+        TODO: Build in a relationship between larval_cohort mass and adult cohort mass.
+        TODO: Is adult_mass the correct mass threshold?
+        TODO: If the time step drops below a month, this needs an intermediary stage.
+
+        Args:
+            larval_cohort: The cohort in its larval stage to be transformed.
+        """
+
+        # inflict a mortality
+        number_dead = ceil(
+            larval_cohort.individuals * larval_cohort.constants.metamorph_mortality
+        )
+        larval_cohort.die_individual(number_dead, self.carcass_pool)
+        # collect the adult functional group
+        adult_functional_group = get_functional_group_by_name(
+            self.functional_groups,
+            larval_cohort.functional_group.offspring_functional_group,
+        )
+        # create the adult cohort
+        adult_cohort = AnimalCohort(
+            adult_functional_group,
+            adult_functional_group.birth_mass,
+            0.0,
+            larval_cohort.individuals,
+            self.constants,
+        )
+
+        # add a new cohort of the parental type to the community
+        self.animal_cohorts[adult_cohort.name].append(adult_cohort)
+
+        # remove the larval cohort
+        larval_cohort.is_alive = False
+        self.remove_dead_cohort(larval_cohort)
+
+    def metamorphose_community(self) -> None:
+        """Handle metamorphosis for all applicable cohorts in the community."""
+
+        for cohort in list(self.all_animal_cohorts):
```

Comment:
Do you need to convert this to a `list`? Would this work:

```suggestion
        for cohort in self.all_animal_cohorts:
```